.circleci/config.yml

```diff
@@ -1,14 +1,194 @@
+          command: docker build . -f Dockerfile -t ${REPONAME}/${FRONTEND_IMAGE}:${IMGTAG}
+          working_directory: ~/project/litmus-portal/frontend
+      - run:
+          name: Save frontend docker image
+          command: docker save -o workspace/frontend.tar ${REPONAME}/${FRONTEND_IMAGE}:${IMGTAG}
+      - persist_to_workspace:
+          root: workspace
+          paths:
+            - ${GRAPHQL_SERVER_IMAGE}.tar
+            - ${SUBSCRIBER_IMAGE}.tar
+            - ${SELF_DEPLOYER_IMAGE}.tar
+            - ${AUTHENTICATION_SERVER_IMAGE}.tar
+            - ${FRONTEND_IMAGE}.tar
+
+  docker-push:
     machine:
-      image: ubuntu-1604:201903-01
-    working_directory: ~/go/src/github.com/litmuschaos/litmus
+      image: circleci/classic:201808-01
     environment:
-      GO111MODULE: "on"
+      IMGTAG: ci
     steps:
-      - run : |
-          cd && source /opt/circleci/.nvm/nvm.sh && nvm install v10 && nvm alias default 10 && wget https://dl.google.com/go/go1.14.6.linux-amd64.tar.gz && sudo tar -xvf go1.14.6.linux-amd64.tar.gz && sudo rm -rf /usr/local/go && sudo mv go /usr/local && export GOROOT=/usr/local/go
+      - attach_workspace:
+          at: /tmp/workspace
+      - run: |
+          cat /tmp/workspace/env-vars >> $BASH_ENV
       - checkout
+      - run:
+          name: Loading all docker images
+          command: |
+            docker load -i /tmp/workspace/${GRAPHQL_SERVER_IMAGE}.tar
+            docker load -i /tmp/workspace/${SUBSCRIBER_IMAGE}.tar
+            docker load -i /tmp/workspace/${SELF_DEPLOYER_IMAGE}.tar
+            docker load -i /tmp/workspace/${AUTHENTICATION_SERVER_IMAGE}.tar
+            docker load -i /tmp/workspace/${FRONTEND_IMAGE}.tar
+      - run:
+          name: Pushing graphql server
+          command: bash ./hack/push --TYPE=ci --REPONAME=${REPONAME} --IMGNAME=${GRAPHQL_SERVER_IMAGE} --IMGTAG=${IMGTAG}
+      - run:
+          name: Pushing frontend server
+          command: bash ./hack/push --TYPE=ci --REPONAME=${REPONAME} --IMGNAME=${FRONTEND_IMAGE} --IMGTAG=${IMGTAG}
+      - run:
+          name: Pushing auth server
+          command: bash ./hack/push --TYPE=ci --REPONAME=${REPONAME} --IMGNAME=${AUTHENTICATION_SERVER_IMAGE} --IMGTAG=${IMGTAG}
+      - run:
+          name: Pushing subscriber server
+          command: bash ./hack/push --TYPE=ci --REPONAME=${REPONAME} --IMGNAME=${SUBSCRIBER_IMAGE} --IMGTAG=${IMGTAG}
+      - run:
+          name: Pushing self deployer
+          command: bash ./hack/push --TYPE=ci --REPONAME=${REPONAME} --IMGNAME=${SELF_DEPLOYER_IMAGE} --IMGTAG=${IMGTAG}
+
+  release:
+    machine:
+      image: circleci/classic:201808-01
+    environment:
+      IMGTAG: ci
+    steps:
+      - attach_workspace:
+          at: /tmp/workspace
       - run: |
-          source /opt/circleci/.nvm/nvm.sh && nvm alias default 10 && cd litmus-portal && make all
\ No newline at end of file
+          cat /tmp/workspace/env-vars >> $BASH_ENV
```

Comment:
here also source is required

Reply:
Here env will store temporarily. the source is required when we have to switch the session

Reply:
but without source this shell will not be able to understand any envs which are stored in BASH_ENV(this is just a variable)

Reply:
Got your point, But jobs are able to read the env without source

Reply:
okay! interesting